test/teos/e2e/test_cli_e2e.py

```diff
@@ -0,0 +1,117 @@
+    watching = all_appointments.get("watcher_appointments")
+    responding = all_appointments.get("responder_trackers")
+    assert len(watching) == 0 and len(responding) == 0
+
+    # Register a user
+    available_slots, subscription_expiry = teos_client.register(user_id, teos_id, teos_base_endpoint)
+
+    # After that we can build an appointment and send it to the tower
+    commitment_tx, commitment_txid, penalty_tx = create_txs()
+    appointment_data = build_appointment_data(commitment_txid, penalty_tx)
+    appointment = teos_client.create_appointment(appointment_data)
+    add_appointment(teos_id, appointment)
+
+    # Now there should now be one appointment in the watcher
+    all_appointments = json.loads(rpc_client.get_all_appointments())
+    watching = all_appointments.get("watcher_appointments")
+    responding = all_appointments.get("responder_trackers")
+    assert len(watching) == 1 and len(responding) == 0
+
+    # Trigger a breach and check again; now the appointment should be in the responder
+    generate_block_with_transactions(commitment_tx)
+    sleep(1)
+
+    all_appointments = json.loads(rpc_client.get_all_appointments())
+    watching = all_appointments.get("watcher_appointments")
+    responding = all_appointments.get("responder_trackers")
+    assert len(watching) == 0 and len(responding) == 1
+
+    # Now let's mine some blocks so the appointment reaches its end. We need 100 + EXPIRY_DELTA -1
+    generate_blocks(100 + config.get("EXPIRY_DELTA"))
+    sleep(1)
+
+    # Now the appointment should not be in the tower, back to 0
+    all_appointments = json.loads(rpc_client.get_all_appointments())
+    watching = all_appointments.get("watcher_appointments")
+    responding = all_appointments.get("responder_trackers")
+    assert len(watching) == 0 and len(responding) == 0
+
+
+def test_get_tower_info(teosd, rpc_client):
+    tower_info = json.loads(rpc_client.get_tower_info())
+    assert set(tower_info.keys()) == set(
+        ["n_registered_users", "tower_id", "n_watcher_appointments", "n_responder_trackers"]
+    )
+
+
+def test_get_users(teosd, rpc_client):
+    _, teos_id = teosd
+
+    # Register a user
+    available_slots, subscription_expiry = teos_client.register(user_id, teos_id, teos_base_endpoint)
+
+    users = json.loads(rpc_client.get_users())
+
+    assert user_id in users
+
+
+def test_get_user(teosd, rpc_client):
+    _, teos_id = teosd
+
+    # Register a user
+    available_slots, subscription_expiry = teos_client.register(user_id, teos_id, teos_base_endpoint)
+
+    # Get back its info
+    user = json.loads(rpc_client.get_user(user_id))
+
+    assert set(user.keys()) == set(["appointments", "available_slots", "subscription_expiry"])
+
+
+def test_get_user_non_existing(teosd, rpc_client):
+    # Get a user that does not exist
+    with pytest.raises(InvalidParameter):
+        json.loads(rpc_client.get_user("00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff00"))
```

Comment:
What's the point of json.loads here if `rpc_client.get_user` will fail?

Reply:
Indeed, removed in 37ef745fe6a74b3cdad0a95c96b029c4ec59e57f.